### ISSUE TITLE
ts_web_test: use headless chrome when not debugging

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/rules_typescript#104]
in google3 we use headless chrome
via https://github.com/karma-runner/karma-chrome-launcher#headless-chromium-with-puppeteer

when the user does `ibazel run :my_web_test`
because ibazel is the reporter.

When run with `bazel test` or `ibazel test`, then we do open the window, assuming that users are running the test only once so they would like to debug.

/cc @vikerman

**Comment by alexeagle** [comment]
actually we should print a warning if you do `ibazel test` since it won't do the `ibazel` thing

**Comment by yohcop** [comment]
In my opinion, it would be nicer if `bazel test` would not open a window actually. To make it easier to run things like `bazel test //...` to test a whole workspace (with non typescript tests as well), maybe in some continuous deployment setup.

**Comment by yohcop** [comment]
After looking more at the code, it seems that if the DISPLAY environment variable is not set, headless chrome will be used by default. I guess I can work with that.

**Comment by alexeagle** [comment]
I did that because it was convenient - but @mhevery found that on Mac that environment variable isn't set so you get headless chrome for `bazel run` as well. You can still manually connect a browser to Karma in this case, so it's not exactly broken - but the inconsistency will make this harder to document.

**Comment by olore** [comment]
On OSX, when I run `bazel test //sometest` headless is used, and when I run `bazel run //sometest` Chrome pops open. That feels correct to me. Maybe this is resolved?

**Comment by alexeagle** [comment]
@vikerman what do you think, should we re-visit this behavior or just let Bazel and Blaze be different in this regard?

**Comment by KrauseStefan** [comment]
Adding to the conversation
The DISPLAY environment variable does not exist on windows.
Having a way to easily switch between headless and normal from the command line is needed I think.

I currently have 
``` powershell
# Allow running karma in non headless mode
run --define DISPLAY=true
```
When I run my test with `bazel run :test` I get a window that I can debug with
When I run with `bazel  test :test` the test just run headless and I get the results

Having a similar way to control whether external debugging is enabled or not would be nice.

**Comment by alexeagle** [comment]
We already recommend the `--config=debug` pattern for other debugging scenarios in nodejs tests. Maybe we should expand this to trigger lots of debug scenarios.
